stripe/src/main/java/com/stripe/android/util/DateUtils.java

```diff
@@ -37,6 +45,89 @@ public static boolean hasMonthPassed(int year, int month) {
+    }
+
+    /**
+     * Checks whether or not the input month and year has yet expired.
+     *
+     * @param expiryMonth An integer representing a month. Only values 1-12 are valid,
+     *                    but this is called by user input, so we have to check outside that range.
+     * @param expiryYear An integer representing the full year (2017, not 17). Only positive values
+     *                   are valid, but this is called by user input, so we have to check outside
+     *                   for otherwise nonsensical dates. This code cannot validate years greater
+     *                   than {@link #MAX_VALID_YEAR 9980} because of how we parse years in
+     *                   {@link #convertTwoDigitYearToFour(int, Calendar)}.
+     * @return {@code true} if the current month and year is the same as or later than the input
+     * month and year, {@code false} otherwise. Note that some cards expire on the first of the
+     * month, but we don't validate that here.
+     */
+    public static boolean isExpiryDataValid(int expiryMonth, int expiryYear) {
+        return isExpiryDataValid(expiryMonth, expiryYear, Calendar.getInstance());
+    }
+
+    @VisibleForTesting
+    static boolean isExpiryDataValid(int expiryMonth, int expiryYear, @NonNull Calendar calendar) {
+        if (expiryMonth < 1 || expiryMonth > 12) {
+            return false;
+        }
+
+        if (expiryYear < 0 || expiryYear > MAX_VALID_YEAR) {
+            return false;
+        }
+
+        int currentYear = calendar.get(Calendar.YEAR);
+        if (expiryYear < currentYear) {
+            return false;
+        } else if (expiryYear > currentYear) {
+            return true;
+        } else { // the card expires this year
+            int readableMonth = calendar.get(Calendar.MONTH) + 1;
+            return expiryMonth >= readableMonth;
+        }
+    }
+
+    /**
+     * Converts a two-digit input year to a four-digit year. As the current calendar year
+     * approaches a century, we assume small values to mean the next century. For instance, if
+     * the current year is 2090, and the input value is "18", the user probably means 2118,
+     * not 2018. However, in 2017, the input "18" probably means 2018. This code should be
+     * updated before the year 9981.
+     *
+     * @param inputYear a two-digit integer, between 0 and 99, inclusive
+     * @return a four-digit year
+     */
+    @IntRange(from = 1000, to = 9999)
+    public static int convertTwoDigitYearToFour(@IntRange(from = 0, to = 99) int inputYear) {
+        return convertTwoDigitYearToFour(inputYear, Calendar.getInstance());
+    }
+
+    @VisibleForTesting
+    @IntRange(from = 1000, to = 9999)
+    static int convertTwoDigitYearToFour(
+            @IntRange(from = 0, to = 99) int inputYear,
+            @NonNull Calendar calendar) {
+        int year = calendar.get(Calendar.YEAR);
+        // Intentional integer division
+        int centuryBase = year / 100;
```

Comment:
I made them up -- the idea (I didn't see much other guidance, and I'm willing to change) is that, if the year is 2010, and you say "98", you probably mean 1998. But if the year is 2070, you mean 2098. Also, if the year is 2094, and you say "07", you mean "2107," but if you say "07" today, you mean 2007.

It's not really something to worry about, given that the code will do approximately nothing for about sixty years (though for the next three years, it will interpret 80-99 as 1980-1999)